storage/innobase/row/row0ins.cc

```diff
@@ -2619,6 +2619,22 @@ static uint64_t row_parse_int(const byte *data, size_t len,
   return 0;
 }
 
+inline bool dict_table_t::can_bulk_insert(const trx_t *trx) noexcept
+{
+  if (is_temporary() || versioned() || has_spatial_index())
+    return false;
+  /* Bulk insert is not compatible with HA_CHECK_UNIQUE_AFTER_WRITE.
+  Refuse bulk insert if HA_KEY_ALG_LONG_HASH indexes exist.
+  handler::ha_check_long_uniques() assumes that all data
+  passed to ha_innobase::write_row() is available immediately. */
+  if (const char *s= v_col_names)
+    for (auto n= n_v_cols; n--; s+= strlen(s) + 1)
+      if (!strncmp(s, C_STRING_WITH_LEN("DB_ROW_HASH_")))
+        return false; /* make_long_hash_field_name() */
+  return !trx->check_foreigns || (foreign_set.empty()
+                                  && referenced_set.empty());
```

Comment:
In my suggestion, `trx` was a reference, and the statement would fit in a single line that is just at our 79-character limit:
```c++
  return !trx.check_foreigns || (foreign_set.empty() && referenced_set.empty());
```
In the TAB indented old InnoDB formatting style, lines were split before binary operators such as `&&`. The common formatting style is to split lines after a binary operator.